cpp/CMakeLists.txt

```diff
@@ -131,7 +131,23 @@ include(cmake/thirdparty/get_thread_pool.cmake)
 # ##################################################################################################
 # * library targets --------------------------------------------------------------------------------
 
-set(SOURCES "src/file_handle.cpp")
+set(SOURCES
+    "src/batch.cpp"
+    "src/bounce_buffer.cpp"
+    "src/buffer.cpp"
+    "src/cufile/config.cpp"
+    "src/cufile/driver.cpp"
+    "src/defaults.cpp"
+    "src/error.cpp"
+    "src/file_handle.cpp"
+    "src/posix_io.cpp"
+    "src/shim/cuda.cpp"
+    "src/shim/cufile.cpp"
+    "src/shim/libcurl.cpp"
```

Comment:
I think this should only be built if `CUDF_KVIKIO_REMOTE_IO=ON`. See https://github.com/NVIDIA/spark-rapids-jni/issues/2766.

Reply:
The option I am looking for may already exist: `KvikIO_REMOTE_SUPPORT`